### DISCUSSION TITLE
Unable to load data

### DISCUSSION THREAD

**Discussion by yangyangdotcom** [pyg-team/pytorch_geometric#5574] [Q&A] [answered]
I was having the following error when I try to load my data for training

`RuntimeError: The 'data' object was created by an older version of PyG. If this error occurred while loading an already existing dataset, remove the 'processed/' directory in the dataset's root folder and try again.`

I am new to this , any helps would be much appreciated

Edit:

To make my question clear, the following is my code

Code:
```
def __getitem__(self, index):
      
      prot_1 = os.path.join(self.processed_dir, self.protein_1[index]+".pt")
      prot_2 = os.path.join(self.processed_dir, self.protein_2[index]+".pt")
      prot_1 = torch.load(glob.glob(prot_1)[0])
      print(prot_1)
      prot_2 = torch.load(glob.glob(prot_2)[0])
      print("stops here")
      return prot_1, prot_2, torch.tensor(self.label[index])
```

It is at the `print(port_1)` where the compiler screams error
The `self.processed_dir` will go to the data folder named "processed" (this is the folder that I created myself) to grab all the '.pt' files

**Comment by EdisonLeeeee** [comment, marked as answer]
You can simply follow the instruction and "remove the 'processed/' directory in the dataset's root folder and try again".

**Reply by yangyangdotcom** [reply]
But I do not know where to find the processed/ directory. At first I though it wants me to remove my dataset folder so I delete it even though it doesn't make sense to me. Turns out, it is not my dataset folder that causing this error.

**Reply by EdisonLeeeee** [reply]
The processed/ directory is located in your dataset's `root` folder, where `root` is specified when you initialize the dataset. For example:
```python
dataset = Reddit('data/')
```
Then your processed dir is located at `data/processed`

**Reply by yangyangdotcom** [reply]
Thank you for your reply. So the processed folder is generated by the pyG? I cannot find it in my dataset's root folder

**Comment by EdisonLeeeee** [comment]
That's weird. Could you show me your code to figure it out?

**Reply by yangyangdotcom** [reply]
I was trying to replicate [this ](https://github.com/JhaKanchan15/PPI_GNN)project. It has a processed file that contains all the clean data. When I was trying follow the README.md and run the train.py, the Runtime error occurs. I believe it is the `__getitem__` from `data_prepare.py`  causing this problem. I hope this helps you better understand my problem. Thank you

**Reply by rusty1s** [reply]
What does `print(self.processed_dir)` output? This is the folder that you need to remove.

**Reply by yangyangdotcom** [reply]
It stores my dataset. If I remove it, there will no dataset to train my model

**Reply by EdisonLeeeee** [reply]
It won't. Your raw dataset is saved at `/raw` dir and the files in `processed/` dir were generated by processing on it.

**Reply by yangyangdotcom** [reply]
Sorry I was gonna try the solution you provide but somehow I got another problem which is segfault and:
`libc++abi: terminating with uncaught exception of type std::length_error: vector
zsh: abort`
This happen when I try to import torch_geometric
I will try your solution once I resolve my problem. Thank you!

**Reply by yangyangdotcom** [reply]
What does the processed/ file actually meant to store? Is it used to store data that is after pre-processing or data that has been feed into the model?

**Reply by rusty1s** [reply]
`raw` refers to your raw data in any format, `processed` refers to the processed data in PyG format, i.e., `Data` or `HeteroData` holding PyTorch tensors. It refers to the data after pre-processing/that is inputted into your model.

**Reply by yangyangdotcom** [reply]
So after I finish the pre-processing phase and got the pre-processed data saved in the processed/ directory, can I use it to as an input to my machine learning model? Or I should move it to other folder?

**Reply by rusty1s** [reply]
Yes, you can directly make use of it now.

**Reply by yangyangdotcom** [reply]
I think my problem on this topic is fixed. Thank you all for the help. Much appreciated
